source_code/main_mcu/src/COMMS/comms_aux_mcu.c

```diff
@@ -244,7 +245,94 @@ void comms_aux_mcu_deal_with_received_event(aux_mcu_message_t* received_message)
+    send_message->FIDO2_message.message_type = AUX_MCU_FIDO2_AUTH_CRED_RSP;
+    send_message->payload_length1 = sizeof(fido2_message_t);
+    return FIDO2_MSG_RCVD;
+}
+
+/*! \fn     comms_aux_mcu_handle_FIDO2_make_auth_data_msg(fido2_message_t const *received_message, aux_mcu_message_t *send_message)
+*   \brief  routine handling authenticating a credential
+*   \param  received_message The received message
+*   \param  send_message The response message
+*   \return FIDO2_MSG_RCVD
+*/
+static comms_msg_rcvd_te comms_aux_mcu_handle_FIDO2_make_auth_data_msg(fido2_message_t const *received_message, aux_mcu_message_t *send_message)
+{
+    FIDO2_make_auth_data_req_message_t const *request = &received_message->FIDO2_make_auth_data_req_message;
+    FIDO2_make_auth_data_rsp_message_t *response = &send_message->FIDO2_message.FIDO2_make_auth_data_rsp_message;
+
+    fido2_process_make_auth_data(request, response);
+
+    send_message->message_type = AUX_MCU_MSG_TYPE_FIDO2;
+    send_message->FIDO2_message.message_type = AUX_MCU_FIDO2_MAD_RSP;
+    send_message->payload_length1 = sizeof(fido2_message_t);
+    return FIDO2_MSG_RCVD;
+}
+
+/*! \fn     comms_aux_mcu_handle_FIDO2_unknown_msg(fido2_message_t const *received_message, aux_mcu_message_t *send_message)
+*   \brief  routine handling unknown FIDO2 messages
+*   \param  received_message The received message
+*   \param  send_message The response message
+*   \return UNKNOW_MSG_RCVD
+*/
+static comms_msg_rcvd_te comms_aux_mcu_handle_FIDO2_unknown_msg(fido2_message_t const *received_message, aux_mcu_message_t *send_message)
+{
+    return UNKNOW_MSG_RCVD;
+}
+
+/*! \fn     comms_aux_mcu_handle_FIDO2_message(fido2_message_t *received_message)
+*   \brief  routine dealing with FIDO2 messages
+*   \param  received_message The received message
+*   \return FIDO2_MSG_RCVD or UNKNOW_MSG_RCVD
+*/
+static comms_msg_rcvd_te comms_aux_mcu_handle_FIDO2_message(fido2_message_t const *received_message)
+{
+    comms_msg_rcvd_te msg_rcvd = UNKNOW_MSG_RCVD;
+    uint16_t message_type = received_message->message_type;
+
+    if (message_type >= AUX_MCU_MSG_TYPE_FIDO2_START && message_type <= AUX_MCU_MSG_TYPE_FIDO2_END)
+    {
+        switch (message_type) {
+            case AUX_MCU_FIDO2_AUTH_CRED_REQ:
+                msg_rcvd = comms_aux_mcu_handle_FIDO2_auth_cred_msg(received_message, &aux_mcu_send_message);
+                break;
+            case AUX_MCU_FIDO2_MAD_REQ:
+                msg_rcvd = comms_aux_mcu_handle_FIDO2_make_auth_data_msg(received_message, &aux_mcu_send_message);
+                break;
+            case AUX_MCU_FIDO2_AUTH_CRED_RSP:
+            case AUX_MCU_FIDO2_MAD_RSP:
+                msg_rcvd = comms_aux_mcu_handle_FIDO2_unknown_msg(received_message, &aux_mcu_send_message);
+                break;
+            default:
+                msg_rcvd = UNKNOW_MSG_RCVD;
+                break;
+        };
+    }
+
+    /* Send message */
+    comms_aux_mcu_send_message(FALSE);
```

Comment:
do you maybe want to include this function call in the sanity check if in case you receive an erroneous fido2 message?

Reply:
Can you explain more what you mean?

Reply:
I simply meant putting  comms_aux_mcu_send_message(FALSE); in your if() :)